database/mysql.go

```diff
@@ -148,37 +178,61 @@ func CreateTestMySQLDB(t *testing.T) *TestMySQLDB {
 		t.Skip("Docker not enabled")
 	}
 
-	config, container, err := RunMySQLDockerInstance(&DatabaseConfig{})
-	if err != nil {
-		t.Fatal(err)
+	err := SharedMySQL.Start()
+	require.NoError(t, err)
+
+	dbName, err := CreateTemporaryDatabase(SharedMySQL.Config)
+	require.NoError(t, err)
+
+	dbConfig := &DatabaseConfig{
+		DatabaseName: dbName,
+		MySQL:        SharedMySQL.Config.MySQL,
 	}
 
 	logger := log.NewNopLogger()
-
 	ctx, cancelFunc := context.WithCancel(context.Background())
-
-	db, err := NewAndMigrate(ctx, logger, *config)
+	db, err := NewAndMigrate(ctx, logger, *dbConfig)
 	if err != nil {
-		container.Close()
 		t.Fatal(err)
 	}
 
 	// Don't allow idle connections so we can verify all are closed at the end of testing
 	db.SetMaxIdleConns(0)
 
 	return &TestMySQLDB{
-		DB:        db,
-		container: container,
-		shutdown:  cancelFunc,
-		t:         t,
+		DB:       db,
+		name:     dbName,
+		shutdown: cancelFunc,
+		t:        t,
 	}
 }
 
-func RunMySQLDockerInstance(config *DatabaseConfig) (*DatabaseConfig, *dockertest.Resource, error) {
-	if config.DatabaseName == "" {
-		config.DatabaseName = "test"
+// We connect as root to MySQL server and create database with random name to
+// run our migrations on it later.
+func CreateTemporaryDatabase(config *DatabaseConfig) (string, error) {
```

Comment:
Since this function should only be called in tests, we can pass in `testing.T` and fail the test if there is an error.
Also, we don't need to pass in the db config, we can define it in here since we know what it is.

The function signature should look like:
`func CreateTemporaryDatabase(t *testing.T) (string) {...}`

Reply:
I prefer to not pass any dependencies (t) if we can avoid it and it does not give us any benefit. Without this dependency this function can be used in different contexts and methods that may not have `t`.

Reply:
Hmm I don't know if we'll ever use this outside of `testing.T` though; are there any use cases you're thinking of? And since it's part of the API, it's probably better that it depends on `testing.T` so it can only be used in that context.

Reply:
Right now it's a tiny simple function that creates temporary DB based on provided config. That's it.
I'll make it private, but I don't see why we should accept `t`. I definitely had issues with methods that accepts `t` in our code base when I wanted to use them in a different context. But unfortunately I can't tell you right now what was the issue...

Reply:
Passing `testing.T` to test helpers is a very common testing pattern in Go. It's handy because you can call `t.Fatal` and the test will quit rather than checking the returned error.